fontc/src/timing.rs

```diff
@@ -4,7 +4,44 @@
 
 use fontbe::orchestration::{AnyWorkId, WorkId as BeWorkIdentifier};
 use fontir::orchestration::WorkId as FeWorkIdentifier;
-use std::{collections::HashMap, io, thread::ThreadId, time::Instant};
+use std::{
+    collections::HashMap,
+    io,
+    thread::ThreadId,
+    time::{Duration, Instant},
+};
+
+/// A trait for clock implementations, allowing for different time sources
+/// such as real time or fake time for testing.
+pub trait Clock: std::fmt::Debug + Clone + 'static {
```

Comment:
I really don't love using a trait for this, since it leaks this implementation detail out into the rest of the code... but I guess rod didn't like the other way? not a hill I need to die on.